applications/MeshingApplication/custom_utilities/mmg_utilities.h

```diff
@@ -0,0 +1,645 @@
+// KRATOS  __  __ _____ ____  _   _ ___ _   _  ____
+//        |  \/  | ____/ ___|| | | |_ _| \ | |/ ___|
+//        | |\/| |  _| \___ \| |_| || ||  \| | |  _
+//        | |  | | |___ ___) |  _  || || |\  | |_| |
+//        |_|  |_|_____|____/|_| |_|___|_| \_|\____| APPLICATION
+//
+//  License:		 BSD License
+//                       license: MeshingApplication/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#if !defined(KRATOS_MMG_UTILITIES)
+#define KRATOS_MMG_UTILITIES
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "meshing_application.h"
+#include "includes/key_hash.h"
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+
+// NOTE: The following contains the license of the MMG library
+/* =============================================================================
+**  Copyright (c) Bx INP/Inria/UBordeaux/UPMC, 2004- .
+**
+**  mmg is free software: you can redistribute it and/or modify it
+**  under the terms of the GNU Lesser General Public License as published
+**  by the Free Software Foundation, either version 3 of the License, or
+**  (at your option) any later version.
+**
+**  mmg is distributed in the hope that it will be useful, but WITHOUT
+**  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+**  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+**  License for more details.
+**
+**  You should have received a copy of the GNU Lesser General Public
+**  License and of the GNU General Public License along with mmg (in
+**  files COPYING.LESSER and COPYING). If not, see
+**  <http://www.gnu.org/licenses/>. Please read their terms carefully and
+**  use this copy of the mmg distribution only if you accept them.
+** =============================================================================
+*/
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+    /// Index definition
+    typedef std::size_t                                               IndexType;
+
+    /// Size definition
+    typedef std::size_t                                                SizeType;
+
+    /// Index vector
+    typedef std::vector<IndexType>                              IndexVectorType;
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class MmgUtilities
+ * @ingroup MeshingApplication
+ * @brief This class are different utilities that uses the MMG library
+ * @details This class are different utilities that uses the MMG library
+ * @author Vicente Mataix Ferrandiz
+ */
+template<MMGLibray TMMGLibray>
+class KRATOS_API(MESHING_APPLICATION) MmgUtilities
```

Comment:
I think `KRATOS_API(MESHING_APPLICATION)` goes with the instantiation rather than here.

Reply:
I think not, @roigcarlo ?